include/openthread/platform/crypto.h

```diff
@@ -0,0 +1,537 @@
+ *
+ */
+
+#ifndef OT_PLAT_CRYPTO_KEY_ATTRIBUTES_SIZE
+#define OT_PLAT_CRYPTO_KEY_ATTRIBUTES_SIZE 1
+#endif
+
+/**
+ * This enumeration defines the crypto to use.
+ *
+ */
+typedef enum
+{
+    OT_CRYPTO_TYPE_USE_LITERAL_KEYS = 0, ///< Use Literal Keys.
+    OT_CRYPTO_TYPE_USE_KEY_REFS     = 1, ///< Use Key References.
+} otCryptoType;
+
+/**
+ * This enumeration defines the key types.
+ *
+ */
+typedef enum
+{
+    OT_CRYPTO_KEY_TYPE_RAW,  ///< Key Type: Raw Data.
+    OT_CRYPTO_KEY_TYPE_AES,  ///< Key Type: AES.
+    OT_CRYPTO_KEY_TYPE_HMAC, ///< Key Type: HMAC.
+} otCryptoKeyType;
+
+/**
+ * This enumeration defines the key algorithms.
+ *
+ */
+typedef enum
+{
+    OT_CRYPTO_KEY_ALG_VENDOR,       ///< Key Algorithm: Vendor Defined.
+    OT_CRYPTO_KEY_ALG_AES_ECB,      ///< Key Algorithm: AES ECB.
+    OT_CRYPTO_KEY_ALG_HMAC_SHA_256, ///< Key Algorithm: HMAC SHA-256.
+} otCryptoKeyAlgorithm;
+
+/**
+ * This enumeration defines the key usage types.
+ *
+ */
+typedef enum
+{
+    OT_CRYPTO_KEY_USAGE_EXPORT    = 1, ///< Key Usage: Key can be exported.
+    OT_CRYPTO_KEY_USAGE_ENCRYPT   = 2, ///< Key Usage: Vendor Defined.
+    OT_CRYPTO_KEY_USAGE_DECRYPT   = 4, ///< Key Usage: AES ECB.
+    OT_CRYPTO_KEY_USAGE_SIGN_HASH = 8, ///< Key Usage: HMAC SHA-256.
+} otCryptoKeyUsage;
+
+/**
+ * This enumeration defines the key storage types.
+ *
+ */
+typedef enum
+{
+    OT_CRYPTO_KEY_STORAGE_VOLATILE,   ///< Key Persistence: Key is volatile.
+    OT_CRYPTO_KEY_STORAGE_PERSISTENT, ///< Key Persistence: Key is persistent.
+} otCryptoKeyStorage;
+
+/**
+ * This datatype represents the key reference.
+ *
+ */
+typedef uint32_t otCryptoKeyRef;
+
+/**
+ * @struct otCryptoKey
+ *
+ * This structure represents the Key Material required for Crypto operations.
+ *
+ */
+
+struct otCryptoKey
+{
+    const uint8_t *mKey;       ///< Pointer to the buffer containing key. NULL indicates to use `mKeyRef`.
+    uint16_t       mKeyLength; ///< The key length in bytes (applicable when `mKey` is not NULL).
+    uint32_t       mKeyRef;    ///< The PSA key ref (requires `mKey` to be NULL).
+};
+
+/**
+ * @struct struct otCryptoKeyAttributes
+ *
+ * This structure represents the Key Attributes structure.
+ *
+ */
+
+struct otCryptoKeyAttributes
+{
+    uint8_t m8[OT_PLAT_CRYPTO_KEY_ATTRIBUTES_SIZE];
+};
+
+/**
+ *
+ * This structure represents the Key key Attributes structure.
+ *
+ */
+
+typedef struct otCryptoKeyAttributes otCryptoKeyAttributes;
+
+/**
+ *
+ * This structure represents the Key Material required for Crypto operations.
+ *
+ */
+typedef struct otCryptoKey otCryptoKey;
+
+/**
+ * Initialise the Crypto module.
+ *
+ * @retval OT_ERROR_NONE          Successfully initialised Crypto module.
+ * @retval OT_ERROR_FAILED        Failed to initialise Crypto module.
+ *
+ */
+otError otPlatCryptoInit(void);
+
+/**
+ * Import a key into PSA ITS.
+ *
+ * @param[out]  aKeyId            Reference to the key to be used for crypto operations.
```

Comment:
As we discussed in the previous comment, can this be inout, as for Non-Volatile keys we will not generate a new key ids.